ingestion/src/metadata/ingestion/source/storage/s3/metadata.py

```diff
@@ -176,13 +176,46 @@ def yield_create_container_requests(
         yield Either(right=container_request)
         self.register_record(container_request=container_request)
 
+    def get_size(self, bucket_name: str, file_path: str) -> Optional[float]:
+        """
+        Method to get the size of the file
+        """
+        try:
+            file_obj = self.s3_client.head_object(Bucket=bucket_name, Key=file_path)
+            return file_obj["ContentLength"]
```

Comment:
hi is this possible that the key is never there?

Reply:
this is always expected from AWS,